cnf-certification-test/platform/suite_test.go

```diff
@@ -236,3 +238,85 @@ func TestTestTainted(t *testing.T) {
 		assert.Equal(t, tc.expectedGetKernelTaintInfoCalls, len(mockFuncs.GetKernelTaintInfoCalls()))
 	}
 }
+
+//nolint:funlen
+func TestTestContainersFsDiff(t *testing.T) {
+	testCases := []struct {
+		runTestReturn  int
+		expectedResult bool
+	}{
+		{
+			runTestReturn:  testhelper.SUCCESS,
+			expectedResult: true,
+		},
+		{
+			runTestReturn:  testhelper.FAILURE,
+			expectedResult: false,
+		},
+		{
+			runTestReturn:  testhelper.ERROR,
+			expectedResult: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		sharedResult := true
+
+		generateEnv := func() *provider.TestEnvironment {
+			return &provider.TestEnvironment{
+				Containers: []*provider.Container{
+					{
+						Data: &v1.Container{
+							Name: "container1",
+						},
+						Podname:   "debug-pod-01",
+						Namespace: "testNamespace",
+						NodeName:  "worker01",
+					},
+				},
+				DebugPods: map[string]*v1.Pod{
+					"worker01": {
+						Spec: v1.PodSpec{
+							NodeName: "worker01",
+							Containers: []v1.Container{
+								{
+									Name: "container1",
+								},
+							},
+						},
+						ObjectMeta: v1meta.ObjectMeta{
+							Name:      "debug-pod-01",
+							Namespace: "testNamespace",
+						},
+					},
+				},
+				GinkgoFuncs: &tnf.GinkgoFuncsMock{
+					GinkgoAbortSuiteFunc: func(message string, callerSkip ...int) {},
+					GinkgoByFunc:         func(text string, callback ...func()) {},
+					GinkgoFailFunc:       func(message string, callerSkip ...int) {},
+					GinkgoSkipFunc:       func(message string, callerSkip ...int) {},
+				},
+				GomegaFuncs: &tnf.GomegaFuncsMock{
+					GomegaExpectSliceBeNilFunc: func(incomingSlice []string) {
+						if len(incomingSlice) != 0 {
+							sharedResult = false
+						}
+					},
+					GomegaExpectStringNotEmptyFunc: func(incomingStr string) {},
+				},
+			}
+		}
+
+		mockFuncs := &cnffsdiff.FsDiffFuncsMock{
+			RunTestFunc: func(ctx clientsholder.Context) {},
+			GetResultsFunc: func() int {
+				return tc.runTestReturn
+			},
+		}
+
+		testContainersFsDiff(generateEnv(), mockFuncs)
```

Comment:
I am having some doubts about writing these suite level unit test cases. Seems to me like a lot of efforts for not too much benefits. The interface makes the code more difficult to understand in my opinion.
For instance, here we just return true/false in the mocked GetResultsFunc and the RunTestFunc does nothing (because it is already tested in the helper test file). This example, might be an extreme case, but in general I would expect the helper to do most of the work and have the test case describe the workflow of the test somaybe less testing needed there?